app/src/main/java/uk/gov/onelogin/appcheck/FirebaseAppCheck.kt

```diff
@@ -12,21 +13,23 @@ import uk.gov.android.authentication.integrity.appcheck.AppChecker
 import uk.gov.android.authentication.integrity.model.AppCheckToken
 
 class FirebaseAppCheck @Inject constructor(
-    private val appCheckFactory: AppCheckProviderFactory
+    private val appCheckFactory: AppCheckProviderFactory,
+    private val context: Context
 ) : AppChecker {
-    override fun init(
-        context: Context
-    ) {
-        Firebase.initialize(context)
+    private val appCheck = Firebase.appCheck
+
+    init {
         Firebase.appCheck.installAppCheckProviderFactory(
             appCheckFactory
         )
+        Firebase.initialize(context)
```

Comment:
I take it this initialize call is required then?

Reply:
It seems to be so in the documentation [here](https://firebase.google.com/docs/app-check/android/play-integrity-provider#initialize)